### DISCUSSION TITLE
Standalone version

### DISCUSSION THREAD

**Discussion by convinced-prog** [publicsamples/Quetzalcoatl#24] [General]
Any chance of a standalone version of 'Q' or Oi Grandad for that matter.
I would really like to see how they run free of the constraints of a DAW as I'm sick of the damn. dumb fool comments thrown at me by any DAW I've tried them in.
They vary from 'buffer size' to actually not knowing what they are dealing with. I even had an incidence of Oi Grandad running beautifully and producing the most fantastic sounds, invisibly without a gui intrface before locking up.
There does seem to be a problem with the gui and the controls.
Thanks for v 0,7, it's looking good.
Just wish I could actually produce something on it.
All the best!

**Comment by RickTaylor79** [comment]
Yeah, definitely. Good idea. 

The thing is, release days are very time consuming, especially when there are multiple platforms to support and as you can see, Q is getting an update every week or so. A standalone version for each OS just adds to this. However, I will make a point of creating standalone versions for stable releases. Keep an eye for a standalone version of Oi, Grandad! v1 soonish. 

By the way, the buffer size error should be gone as of v0.7. Forgot to mention it in the release notes..

**Reply by convinced-prog** [reply]
Thanks. I do appreciate the extent of my request and the pressure upon you to produce so I won't pester you this time.
Feedback. Well 0.71 was the first successful install so far. No hiccups.
One question. Why are there two .hr1 files when you only install one of them.
I don't see how to get back to the main gui after picking samples, etc?
I wish there was a manual for this as it gets quite complex and as I have had audio and midi issues recently I can't tell if the problem is the software or the hardware.
Anyway it gets better.
Thank you for an exciting program.

**Reply by convinced-prog** [reply]
OK
Just found it. the parameters button!
Takes quite a while to activate everything but what fabulous sounds.
Wish I knew what the hell I was doing..
I suppose i will have to check some of your videos.

**Reply by RickTaylor79** [reply]
Hey, that's great news. We got there in the end :)

The hr flies get split up 'cos Github has a 1gb per file upload limit. Annoying, but it is what it is. 

There's an out of date "quick guide" and some out of date videos floating about, I guess most of it still applies. I suppose a proper manual would be useful.. 

There's a setting that may _somewhat_ improve loading times, depending on your hardware. If you're running on a mechanical HDD, change Streaming (under Settings & Presets) to "Slow HDD".

**Reply by convinced-prog** [reply]
Thanks done that.
Left it running all night and fiddled with all the controls.
The expansion loading confuses me and I haven't loaded the extra sample file yet (don't know how to)
Anyway all's well so far. it will take me months to explore all the possibilities.
I will add all those other packs eventually.
Thanks for your help, and patience.

**Comment by RickTaylor79** [comment]
You could try the recently published windows installer, takes a lot of the pain out of installing as it comes with all the expansions pre-loaded, just download the three files ending with [001](https://github.com/publicsamples/Quetzalcoatl/releases/download/0.7.1/Quetzalcoatl.Win.VST3.Installer.zip.001), [002](https://github.com/publicsamples/Quetzalcoatl/releases/download/0.7.1/Quetzalcoatl.Win.VST3.Installer.zip.002), [003](https://github.com/publicsamples/Quetzalcoatl/releases/download/0.7.1/Quetzalcoatl.Win.VST3.Installer.zip.003).. open the file ending in 001 in [7-zip](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://www.7-zip.org/&ved=2ahUKEwiwzunWvoGIAxUZTkEAHe-8KzUQFnoECAkQAQ&usg=AOvVaw3DQYMmols7YuZnGvUB33w4) and run the the install application. It will place all the files in the correct place.

**Comment by convinced-prog** [comment]
Thanks, I'll check it out.

**Comment by convinced-prog** [comment]
Installation went smoothly with the new Windows method. I assume that all
the files were extracted by the 001 zip file.
Cubase didn't take to it right away as its setup (audio) is too complicated
for my brains,too many stupid questions.
Reaper now is not happy but I suspect there is a residue from previous
attempts lurking somewhere.
Funny enough a browser I rarely use Mixcraft 9 was an instant success.
One silly question. Is the crossfader wheel supposed to change color and
icons? Not complaining as it makes the gui more attractive.
Well that's all for now so I look forward to trying some expansion packs.
All the best

**Comment by convinced-prog** [comment]
Hi
All quiet on the western front.
Have been running 'Q' in savihost.exe with great success and none of the hassles of a DAW. Tremendous sounds being produced.
One question. I see there is a new incarnation available v 0.82.
I think you know what I'm about to say.
What is the installaion for this as all I can see is an Audio. zip.
Please let us know the secret.
Thanks

**Comment by RickTaylor79** [comment]
Hello! 

Ah, it should have been up and working last night, but the last build crashed on _all platforms_ . Got Mac working this morning, Windows and Linux should be imminent. 

You'll like this one, not only is the performance significantly improved, but there is no additional content required to get it making a sound. Also I think it's much prettier.

**Comment by convinced-prog** [comment]
Awesome, thanks for the update!

**Comment by convinced-prog** [comment]
Hi
Up and running.
It does look better. I haven't tried it in a DAW as yet.
The vst host I'm using might not have all the functions or resource connections and the gui isn't fitting correctly but so far so good.
Will take the plunge today if all is well
Thank you

**Comment by convinced-prog** [comment]
Hi
I've just returned to my experiments with Q  ( on Linux Ubuntu 26.04) but i don't see how to install. I can follow most of the instructions but where is the vst file necessary to run this as i can't find one.
Also any chance of a stand alone.
Thanks

**Reply by RickTaylor79** [reply]
Hi! 

The VST is included with the Linux tar archive. It can be a little confusing what actually constitutes a VST on linux, it isn't a single file but a folder (the same is true on other platforms, they just obscure this by making them look like a single file). So in this case, the folder Quetzalcoatl.vst3 is what we're after. 

So for a VST to work, we should end up with something like this:

<img width="1003" height="819" alt="Vst3" src="https://github.com/user-attachments/assets/f8c669cf-3428-4ac7-abbd-7cfd678eec37" />

I've yet to make any kind of official announcement, but work on the open source version of Q has ended. A commercial port, written from the ground up, is currently in development (so no standalone for a bit).  Hopefully we'll have a working alpha sometime this year.  Ultimately this will be available for all platforms. People who have the expanded version of Q will get a license.  Aside from a very significant performance increase, there will be several new features: mod matrix, VA mode, granular mode, dedicated waveshaper, better sample editing, freely routable audio rate modulation (think FM but you can route the oscillator to any parameter) and a bunch more. Here's a little teaser:

<img width="950" height="743" alt="Screenshot 2026-06-18 at 11 52 47" src="https://github.com/user-attachments/assets/8c6e28e1-10a7-43ce-90fc-24697970dda9" />

 For the time being I suggest that people use v0.8.7 as it is the most stable.

**Comment by convinced-prog** [comment]
Thanks Rick.
A stable commercial version would be welcome.
None of my daws was recognizing the .so file so i was totally confused.
i will try v0.8.7 as you suggest. Does this apply to my old Grandad as well?
All the best

**Comment by convinced-prog** [comment]
I wonder now if I made a mistake in embracing linux.
How the hell do you get anything to work properly, even the 'standard' compilation of programs is hit and miss. I've only managed to successfully  compile one audio program, SoundThread, in three years using meson. Cmake was a total disaster.
Sorry to unload on you.
What do i do with .so files as no daw I posses will acknowledge their existence, even your software is denied to me.
I still want to stay away from all those cumbersome commercial workstations and have , in Windows, been pretty successful with vst hosts, Do you know of any linux based hosts?
If i could afford it i would move to Mac as Windows was disastrous but that is as it is.
i am going to try Audacity as a possibility. What do you think

**Reply by RickTaylor79** [reply]
Hey, no worries. I know how frustrating this stuff can be. On the other hand, look, you're successfully compiled audio software on linux, what tiny fraction of the population can make that claim? 

As far as getting VST plugins to work, you don't do anything with .so files. If you have .so files in the root of your VST folder, they will not work.  If you're unsure, send me a screenshot of your vst3 folder and I'll check it out.  I think this is the fundamental issue? not the host(s), which should all work once the plugins are in the right place. 

For standalone hosts on Linux , I am aware of [Carla](https://kx.studio/Applications:Carla) and [Element](https://github.com/kushview/element). Both are fine, though I like Element 'cos it's a modular environment in its own right.

Audacity is fine if you want to edit or multitrack audio, but it only works with FX plugins, instruments will not load.

**Reply by RickTaylor79** [reply]
By the way, the latest commit Oi grandad now has a standalone app, this totally makes sense with this version as it now has a record feature. You can get the Linux build [here](https://github.com/publicsamples/Oi-Grandad/actions/runs/27947276063/artifacts/7790221526). I've not tested this yet, as I am away from the desktop. But the Mac version works as expected. 

If this works for you, there is one step needed to make the record feature and presets work. You need to download the content folder and copy it to Home/.config/modularsamples/ (.config is a hidden folder, so if your in Home, press Ctrl + H to reveal).

You can get the public content [here](https://github.com/publicsamples/Oi-Grandad/releases/download/2.4.0/oi_grandad_content.zip). Or if you prefer, get the lasted Expanded version on Gumroad (you should already have access to this).

**Comment by convinced-prog** [comment]
Hi Rick
Hope you are enjoying the sunshine.
It appears all .so files are not created equal.
I can safely announce that Grandad is running again and much smoother than
before. Well done!
I discovered that writing a .desktop file in Ubuntu using the .iso as
executable produces working shortcuts.
Here is where the old man becomes cantankerous. I can run from a double
click in the file manager and he's happy to comply but the .desktop file
even when set to executable doesn't.
Is there some evil intent by programmers to frustrate a humble soul like
myself?
For goodness ( swear words deleted) what is their problem? What exactly is
an .so file as I can't find a daw or a host program that has even heard of
them. Why are some executable and others shared libraries?
Anyway it's too early in the morning for this as I need my breakfast and I
hear the old man in the kitchen extracting his falsies from his bowl in the
cupboard.
Have a nice day and many thanks for keeping my ageing brain cells in use.

**Comment by RickTaylor79** [comment]
Lovely, isn't it?! Unfortunately my brain stops working in this heat.

I think you need to forget about .so files. They are the linux equivalent of a DLL on windows, which is to say nothing end users need to worry about in 99% of cases. So say we're installing the 0.8.7 version of Q, we'd:

1. Extract the archive
2. Move the "Quetzalcoatl.vst3" folder to Home./vst3/ 
3. Move the content folder, "Quetzalcoatl" to home/.config/modularsamples

As you probably know, when a folder has a full stop before it this means it is hidden, to reveal it press Ctrl + H.

Have a look at this video, it goes through the process of installing plugins on linux:

https://www.youtube.com/watch?v=A3lmVQCy_fc

If you follow this, you'll have working plugins. Promise. I think .so (haaha little nerd joke) anyway.

**Comment by convinced-prog** [comment]
Haha, thanks!
Certainly left me with plenty to think about. This heat is doing my head in.
None of my linux daws can find the 'vst' files for anything I have in my
plugins directory, not
 even .lv1's They are in the searchable dir's, although Ardour found Q' and
then did nothing about it?
Grandad still lurks in the file manager but I want to run 'him' standalone
in the open air.Never mind. Let me go as I'm about to pass out
Will let you know how 'he' gets on.

Need a pint of the 'black stuff' to calm my nerves.

**Comment by convinced-prog** [comment]
What about this Rick. I am still running Grandad from the file manager (
still don't know what the hell I'm doing). Tried to do the same with Q and
would you believe it, it fired up LMMS. Where did that come from? Maybe if
I keep trying I might get a full copy of CuBase.
Still can't figure out how to get things to behave but I will persevere.
All the best
